src/d3a/d3a_core/live_events.py

```diff
@@ -46,6 +49,22 @@ def __init__(self, area_uuid, area_params):
     def apply(self, area):
         if area.uuid != self.area_uuid:
             return False
+        if key_in_dict_and_not_none(self.area_params, 'type'):
+            if area.strategy is None:
+                return False
+            if self.area_params['type'] == "MarketMaker":
+                strategy_obj = MarketMakerStrategy
+                del self.area_params['type']
+                self.area_params.pop('number_of_clones', None)
+                self.area_params.pop('energy_rate_profile_uuid', None)
+                self.area_params = {'strategy_object': strategy_obj(**self.area_params)}
+            elif self.area_params['type'] == "InfiniteBus":
+                strategy_obj = InfiniteBusStrategy
```

Comment:
You can also construct these objects without assigning the class object first if I understand the code correctly.